src/core_modules/capture-core/components/Pages/common/EnrollmentOverviewDomain/EnrollmentPageLayout/hooks/useWidgetColumns.js

```diff
@@ -15,11 +22,19 @@ type Props = {
 const MemoizedWidgets: { [key: string]: React$ComponentType<any> } = {};
 const UnsupportedWidgets: { [key: string]: boolean } = {};
 
+const getPropsForPlugin = ({ program, enrollmentId, teiId, orgUnitId }) => ({
+    programId: program.id,
+    enrollmentId,
+    teiId,
+    orgUnitId,
+});
+
 const renderWidget = (widget: ColumnConfig, availableWidgets, props) => {
```

Comment:
The complexity of `renderWidget` function is quite high at 11. Can this be improved? One suggestion is to split the logic into `renderComponent` and `renderPlugin`.

Reply:
Yes, that makes sense! Extracted this functionality into a separate file and split it up even more. Thanks 👍